tools/libs/light/libxl_pcid.c

```diff
@@ -854,6 +854,47 @@ static void server_fini_one(libxl__gc *gc, struct vchan_client *client)
     free(client);
 }
 
+static bool is_vchan_exist(libxl_ctx *ctx, char *watch_dir)
```

Comment:
This is too complex. Lots of code is copied from the main function.

I played a bit with this. It is much easier to check if `watch_dir` exists. 

Because this is what you are trying to solve: watch fires for a deleted directory, so you can check if directory exists (with call to `xs_directory(.., watch_dir,...)`) and that's all.